core/rawdb/database.go

```diff
@@ -151,12 +151,16 @@ func NewDatabase(db ethdb.KeyValueStore) ethdb.Database {
 // NewDatabaseWithFreezer creates a high level database on top of a given key-
 // value data store with a freezer moving immutable chain segments into cold
 // storage.
-func NewDatabaseWithFreezer(db ethdb.KeyValueStore, freezer string, namespace string, readonly bool) (ethdb.Database, error) {
+func NewDatabaseWithFreezer(db ethdb.KeyValueStore, freezer string, namespace string, args ...bool) (ethdb.Database, error) {
 	// Create the idle freezer instance
-	frdb, err := newFreezer(freezer, namespace, readonly)
+	frdb, err := newFreezer(freezer, namespace, args[0])
 	if err != nil {
 		return nil, err
 	}
+	offset := ReadOffSetOfAncientFreezer(db)
+
+	frdb.offset = offset
```

Comment:
why not read offset in `newFreezer` above

Reply:
This is to assign offset to `newFreezer`, the above `newFreezer` is the initialization without offset.

Reply:
I mean why not init the offset in `newFreezer`

Reply:
because I think`newFreezer` func did not have the parameter of db to read offset through, right?